Src/UIComponents/Overview/Overview.ts

```diff
@@ -0,0 +1,91 @@
+import generateGoalTile from '../GoalTile/GoalTile';
+import { createTileMyWeight } from '../TileMyWeight/TileMyWeight';
+import { createElement, calculateCalories } from '../utils/utils';
+import { User } from '../../../Models/User.model';
+import {saveInLocalStorage, readFromLocalStorage} from '../../Logic/LocalStorage/LocalStorage';
+import generateTileComponent from '../TileComponent/TileComponent';
+
+
+//to call this function (due to the nature of how the gauges are created by the library) please first create a target container in body (make sure it's already in DOM) and only then call the function
+ function overviewComponent(User: User, targetDivClass:string):void{
+    const overviewContainer = createElement('div', 'overview-container') as HTMLDivElement;
+
+    //create a header div
+    const overviewHeader = createElement('div', 'overview-header');
+    const overviewHeaderH1 = createElement('h1', '', 'Overview');
+    const overviewHeaderH2 = createElement('h2', '');
+    overviewHeaderH2.innerHTML = `Hi <span>${User.name}</span>, welcome back!`;
+    overviewHeader.append(overviewHeaderH1, overviewHeaderH2);
+
+    //create the 'goal weight' tile 
+    const goalWeight = createElement('div', 'weight-goal');
+    goalWeight.id = 'overviewGoalWeightTile';
+    const weightTile = generateTileComponent(goalWeight);
+
+    //create the 'today calories' tile
+    const todayCalories = createElement('div', 'today-calories');
+    todayCalories.id = 'overviewTodayCaloriesTile';
+    const todayCaloriesTile = generateTileComponent(todayCalories);
+
+    //create the 'remaining calories' tile and pass the arguments
+    //TO BE ADDED
+
+    //create the weight input tile, save the input to localstorage
+    const myWeightInputTile = createTileMyWeight(User.weights[0].weight, new Date(), User, saveWeightInLocalStorage);
+
+    overviewContainer.append(overviewHeader, weightTile, todayCaloriesTile, myWeightInputTile);
+    document.querySelector(`.${targetDivClass}`).appendChild(overviewContainer);
+    generateGaugesContent(User);
+}
+
+export default overviewComponent;
+
+function getAge(date: Date) {
+    var today = new Date();
+    var birthDate = date;
+    var age = today.getFullYear() - birthDate.getFullYear();
+    var m = today.getMonth() - birthDate.getMonth();
+    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+        age--;
+    }
+    return age;
```

Comment:
all function body could be replaced by one line
return new Date(Date.now() - date.getTime()).getFullYear();

Reply:
nice idea, to be honest I copied the original function from somewhere
one adjustment - I had to subtract 1970 from the result:
return ((new Date(Date.now() - date.getTime()).getFullYear()) - 1970);